aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -259,6 +264,15 @@ impl AptosVM {
         )
         .expect("should be able to create Move VM; check if there are duplicated natives");
 
+        // We use an `Option` to handle the VK not being set on-chain, or an incorrect VK being set
+        // via governance (although, currently, we do check for that in `keyless_account.move`).
+        let pvk = keyless_validation::get_groth16_vk_onchain(resolver)
```

Comment:
this works with the assumption that pvk is the same for the whole block?

i.e. that changing pvk resource, triggers reconfiguration, which will skip all transactions after the change in the block?

@zekun000 , @zjma , @danielxiangzl  - after changes to reconfiguration, is that how things work? or are only feature gates/onchain configs/gas params the ones that are "delayed and applied on epoch change"?

Reply:
i think pvk is treated as an on-chain config too in this PR, and have buffers.

Reply:
So, this is like feature gates/on-chain configs/gas params: changes are delayed and applied on epoch change!

I believe this should be okay for VK change, no?